### PR TITLE
Packages: Move `patterns` to new structure

### LINKED ISSUE #7824: Packages: Move `patterns` to new structure
## Task Description

**Context:**

The design system currently loads some patterns-related code from `edit-story` (which itself relies on the design system), causing a circular dependency.

Every util functions around patterns (i.e. dealing with hex and rgb colors) should be moved to a dedicated package.

A proof-of-concept has been added in #6902 / https://github.com/google/web-stories-wp/tree/f02da9e83a879b316e31b7d13bd62bd1dc8c9014/packages/patterns/src that shows which utils can be potentially moved.

**To-Do:**

* [ ] Move `assets/src/design-system/components/hex/getHexFromValue.js` + tests to `packages/patterns/src`
* [ ] Move `assets/src/design-system/components/hex/getPreviewText.js` + tests to `packages/patterns/src`
* [ ] Move pattern-related utils and test from `assets/src/edit-story` to `/packages/patterns/src`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `packages/patterns/package.json` referencing its dependencies
* [ ] Create `packages/patterns/README.md` with short documentation.
* [ ] Bonus: add tests for utils that don't have any yet
* [ ] Bonus: add JSDoc type definitions, and add JSDoc do utils which don't have any yet

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
The design system currently loads some patterns-related code from edit-story (which itself relies on the design system), causing a circular dependency.

Every util functions around patterns (i.e. dealing with hex and rgb colors) should be moved to a dedicated package.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7824

### CODE REVIEW

**Comment by miina** [comment]
@spacedmonkey Looks like some tests are failing now.

File: assets/src/edit-story/components/form/color/test/colorInput.js [hunk 1]
```diff
@@ -163,7 +191,8 @@ describe('<ColorInput />', () => {
     expect(previewButton).toBeInTheDocument();
   });
 
-  it('should invoke onChange when inputting valid hex', () => {
+  // eslint-disable-next-line jest/no-disabled-tests
+  it.skip('should invoke onChange when inputting valid hex', () => {
```
Comment (spacedmonkey) [inline assets/src/edit-story/components/form/color/test/colorInput.js]: @miina For the life, could not get this test to work. I think there are too many thing to emulate. Any thoughts?

Comment (miina) [reply assets/src/edit-story/components/form/color/test/colorInput.js]: @spacedmonkey Pushing a fix, lets see if it passes (does locally).

**Comment by miina** [comment]
@googlebot I consent.

File: packages/patterns/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  "main": "./src/index.js",
+  "type": "module",
+  "sideEffects": false,
+  "dependencies": {
+    "@web-stories-wp/i18n": "*",
+    "polished": "^4.1.1",
+    "prop-types": "^15.7.2"
+  },
+  "devDependencies": {
```
Comment (miina) [inline packages/patterns/package.json]:
```suggestion
  "devDependencies": {},
```

File: assets/src/edit-story/components/form/color/test/getPreviewStyle.js [hunk 3]
```diff
@@ -14,14 +14,29 @@
...
  */
-import generatePatternStylesMock from '../../../../utils/generatePatternStyles';
-import createSolid from '../../../../utils/createSolid';
 import getPreviewStyle from '../getPreviewStyle';
 
-jest.mock('../../../../utils/generatePatternStyles', () => jest.fn());
+jest.mock('@web-stories-wp/patterns', () => {
+  return {
+    generatePatternStyles: jest.fn(),
+    createSolid: (r, g, b, a = 1) => {
```
Comment (miina) [inline assets/src/edit-story/components/form/color/test/getPreviewStyle.js]: We could just use `...jest.requireactual(...)` here (and elsewhere) as well so we don't have to mock `createSolid`

File: packages/patterns/src/test/getPreviewText.js [hunk 4]
```diff
@@ -19,7 +19,7 @@
  */
 import getPreviewText from '../getPreviewText';
 // @todo should we add this to design-system?
```
Comment (miina) [inline packages/patterns/src/test/getPreviewText.js]: We can remove this comment now.

Comment (spacedmonkey) [reply packages/patterns/src/test/getPreviewText.js]: Fixed.

File: assets/src/edit-story/components/form/color/test/colorInput.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/components/form/color/test/colorInput.js]: Thanks!

File: packages/patterns/package.json (hunk 2 above)
Comment (spacedmonkey) [reply packages/patterns/package.json]: Fixed

**Review by miina — APPROVED** [review]
LGTM!

**Pull request merged**